src/hooks/multiSig/useEligibleSignees.ts

```diff
@@ -1,104 +1,62 @@
-import { Id, type ColonyRole } from '@colony/colony-js';
+import { type ColonyRole } from '@colony/colony-js';
+import { useEffect, useState } from 'react';
 
-import { useMemberContext } from '~context/MemberContext/MemberContext.ts';
-import { type ColonyContributor } from '~types/graphql.ts';
+import { useColonyContext } from '~context/ColonyContext/ColonyContext.ts';
+import {
+  getEligibleSignees as getEligibleSigneesApi,
+  type GetEligibleSigneesResult,
+} from '~utils/multiSig/getEligibleSignees.ts';
 
 interface UseEligibleSigneesParams {
   domainId: number;
-  requiredRoles?: ColonyRole[][];
+  requiredRoles: ColonyRole[];
+}
+
+interface UseEligibleSigneesResult extends GetEligibleSigneesResult {
+  isLoading: boolean;
+  isError: boolean;
 }
 
 export const useEligibleSignees = ({
   domainId,
   requiredRoles,
-}: UseEligibleSigneesParams) => {
-  const { totalMembers } = useMemberContext();
-
-  if (Number.isNaN(domainId)) {
-    throw new Error('domainId must be a number');
-  }
-
-  const domainIds =
-    domainId === Id.RootDomain ? [Id.RootDomain] : [Id.RootDomain, domainId];
-
-  const getEligibleSignees = (members: ColonyContributor[]) => {
-    if (!requiredRoles || requiredRoles.length === 0) {
-      return {
-        eligibleSignees: {},
-        uniqueEligibleSignees: [],
-      };
-    }
-
-    const matches: {
-      [role: number]: { [userAddress: string]: ColonyContributor['user'] };
-    } = {};
-
-    const uniqueEligibleSignees = new Set<ColonyContributor['user']>();
+}: UseEligibleSigneesParams): UseEligibleSigneesResult => {
+  const {
+    colony: { colonyAddress },
+  } = useColonyContext();
+  const [isLoading, setIsLoading] = useState(false);
+  const [isError, setIsError] = useState(false);
+  const [signeesResult, setSigneesResult] = useState<GetEligibleSigneesResult>({
+    countPerRole: {},
+    uniqueEligibleSignees: {},
+    signeesPerRole: {},
+  });
 
-    members.forEach((member) => {
-      if (!member.roles) {
-        return;
+  useEffect(() => {
+    async function fetchEligibleSignees() {
+      try {
+        setIsLoading(true);
+        const response = await getEligibleSigneesApi({
+          requiredRoles,
+          colonyAddress,
+          domainId,
+        });
+        setSigneesResult(response);
+        setIsLoading(false);
+      } catch (error) {
+        console.warn('Error while fetching eligible signees', error);
+        setIsError(true);
+        setIsLoading(false);
       }
+    }
```

Comment:
I would refactor this to make use of the `finally` statement for setting the loading state
```
try {}
catch (error) {}
finally {
   setIsLoading(false);
}
```